packages/core/src/credential/Credential.spec.ts

```diff
@@ -35,87 +38,337 @@ import {
+import * as Attestation from '../attestation'
 
-jest.mock('../attestation/Attestation.chain')
+const rawCType: ICType['schema'] = {
+  $id: 'kilt:ctype:0x2',
+  $schema: 'http://kilt-protocol.org/draft-01/ctype#',
+  title: 'raw ctype',
+  properties: {
+    name: { type: 'string' },
+  },
+  type: 'object',
+}
 
-async function buildCredential(
-  claimer: DidDetails,
-  attesterDid: DidUri,
-  contents: IClaim['contents'],
-  legitimations: ICredential[],
-  sign: SignCallback
-): Promise<ICredential> {
+function buildCredential(
+  claimerDid: DidUri,
+  contents: IClaimContents,
+  legitimations: ICredential[]
+): ICredential {
   // create claim
 
-  const rawCType: ICType['schema'] = {
-    $id: 'kilt:ctype:0x1',
-    $schema: 'http://kilt-protocol.org/draft-01/ctype#',
-    title: 'Credential',
-    properties: {
-      name: { type: 'string' },
-    },
-    type: 'object',
-  }
-
-  const testCType = CType.fromSchema(rawCType)
+  const testCType: ICType = CType.fromSchema(rawCType)
 
-  const claim = Claim.fromCTypeAndClaimContents(
-    testCType,
+  const claim: IClaim = {
+    cTypeHash: testCType.hash,
     contents,
-    claimer.uri
-  )
-  // build request for attestation with legitimations
-  const requestForAttestation = RequestForAttestation.fromClaim(claim, {
+    owner: claimerDid,
+  }
+  // build credential with legitimations
+  const credential = Credential.fromClaim(claim, {
     legitimations,
   })
-  await RequestForAttestation.signWithDidKey(
-    requestForAttestation,
-    sign,
-    claimer,
-    claimer.authenticationKey.id
-  )
-  // build attestation
-  const testAttestation = Attestation.fromRequestAndDid(
-    requestForAttestation,
-    attesterDid
-  )
-  // combine to credential
-  const credential = Credential.fromRequestAndAttestation(
-    requestForAttestation,
-    testAttestation
-  )
   return credential
 }
 
-// Returns a full DID that has the same identifier of the first light DID, but the same key authentication key as the second one, if provided, or as the first one otherwise.
-function createMinimalFullDidFromLightDid(
-  lightDidForId: LightDidDetails,
-  newAuthenticationKey?: DidKey
-): FullDidDetails {
-  const uri = DidUtils.getKiltDidFromIdentifier(
-    lightDidForId.identifier,
-    'full'
-  )
-  const authKey = newAuthenticationKey || lightDidForId.authenticationKey
-
-  return new FullDidDetails({
-    identifier: lightDidForId.identifier,
-    uri,
-    keyRelationships: {
-      authentication: new Set([authKey.id]),
-    },
-    keys: { [authKey.id]: authKey },
+describe('Credential', () => {
+  const identityAlice =
+    'did:kilt:4nv4phaKc4EcwENdRERuMF79ZSSB5xvnAk3zNySSbVbXhSwS'
+  const identityBob =
+    'did:kilt:4s5d7QHWSX9xx4DLafDtnTHK87n5e9G3UoKRrCDQ2gnrzYmZ'
+  let legitimation: ICredential
+
+  beforeEach(async () => {
+    legitimation = buildCredential(identityAlice, {}, [])
   })
-}
 
-describe('RequestForAttestation', () => {
+  it.todo('signing and verification')
+
```

Comment:
I mainly migrated requestForAttestation and credential files, so this is not from me.
The `it.todo` is  reported while doing the tests afaik, so that's not too bad. Best would be to just have tests.